compiler/modules/types/list.bal

```diff
@@ -2,12 +2,16 @@
 
 public type ListAtomicType readonly & record {|
     SemType[] members;
+    // Repeat the last member of `members` this number of times more.
+    // { members: [int], repeatLastMember: 2, rest: NEVER } is same as { members: [int, int, int], repeatLastMember: 0, rest: NEVER }
+    // Must be zero, if members array is empty.
+    int repeatLastMember;
```

Comment:
Looking at the code below, you need often to treat the combination of `members` and `repeatLastMember` as a unit: these two are representing what used to be represented by just members.  So I think everything will be simpler, if you do this:

```
public type ListAtomicTypeMembers readonly & record {|
   SemType[] initial;
   int repeatLastCount;
|};

public type ListAtomicType readonly & record {|
   ListAtomicTypeMembers members;
   SemType rest;
|};
```